hbase-server/src/main/java/org/apache/hadoop/hbase/quotas/QuotaCache.java

```diff
@@ -57,6 +61,11 @@ public class QuotaCache implements Stoppable {
   private static final Logger LOG = LoggerFactory.getLogger(QuotaCache.class);
 
   public static final String REFRESH_CONF_KEY = "hbase.quota.refresh.period";
+
+  // defines the request attribute key which, when provided, will override the request's username
+  // from the perspective of user quotas. This is also the default request attribute key
+  public static final String QUOTA_USER_REQUEST_ATTRIBUTE_OVERRIDE_KEY =
+    "hbase.quota.user.override.key";
```

Comment:
nit: our configuration keys don't generally end with `.key`. Or is the value of `QUOTA_USER_REQUEST_ATTRIBUTE_OVERRIDE_KEY` dereferenced again to look up the actual key to use?

Reply:
Ya I acknowledge this is weird and I'm happy to follow your lead here. I named it `hbase.quota.user.override.key` because its value is a request attribute key that may be provided, rather than the quota user override itself. But I agree it's kind of confusing either way